cmake/toolchain_llvm.cmake

```diff
@@ -0,0 +1,20 @@
+# Copyright 2024 ETH Zurich and University of Bologna.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Viviane Potocnik <vivianep@iis.ee.ethz.ch>
+
+set(CMAKE_EXECUTABLE_SUFFIX ".elf")
+
+set(CMAKE_SYSTEM_NAME Generic)
+
+set(CMAKE_C_COMPILER clang)
+set(CMAKE_CXX_COMPILER clang++)
+set(CMAKE_ASM_COMPILER ${CMAKE_C_COMPILER})
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} --target=riscv32-unknown-elf")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} --target=riscv32-unknown-elf")
+
+set(CMAKE_OBJCOPY llvm-objcopy)
+set(CMAKE_OBJDUMP llvm-objdump)
+set(CMAKE_AR llvm-ar)
```

Comment:
Looks mostly okay (assuming you put into the doc where to get such a marvelous LLVM from / which version you used for testing).
Is CXX-11 required for opentitan?

Reply:
No, the `CXX-11` was indeed an artifact from testing the OpenTitan integration. I added information on which version of LLVM to use to the README. Thanks!

Reply:
I suggest adjusting this to allow specifying a custom location of the toolchain.

```
set(TOOLCHAIN_PREFIX ${TOOLCHAIN_DIR}/bin)
set(LLVM_TAG llvm)

set(CMAKE_C_COMPILER ${TOOLCHAIN_PREFIX}/clang)
set(CMAKE_CXX_COMPILER ${TOOLCHAIN_PREFIX}/clang++)
set(CMAKE_ASM_COMPILER ${TOOLCHAIN_PREFIX}/clang)

set(CMAKE_OBJCOPY ${TOOLCHAIN_PREFIX}/${LLVM_TAG}-objcopy)
set(CMAKE_OBJDUMP ${TOOLCHAIN_PREFIX}/${LLVM_TAG}-objdump)
set(CMAKE_AR ${TOOLCHAIN_PREFIX}/${LLVM_TAG}-ar)
```